### PR TITLE
Resolver 2 and clippy complaints

### PR DESCRIPTION
**Pull request by Grinkers**
While digging around the source code, my editor's configuration was getting very confused and spitting out lots of errors. The core problem was virtual workspace resolver conflicts. Doing `cargo build` (isolated example) outputs this

```
warning: virtual workspace defaulting to `resolver = "1"` despite one or more workspace members being on edition 2021 which implies `resolver = "2"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
note: for more details see https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions
```

I noticed some packages were using 2018 and some were using 2021. I've updated them all to 2021. If you had a differing opinion on MSRV, I also verified `resolver = "1"` works and fixes the original problem.

I have my editor set to be a bit too annoying, perhaps, but it gets me to keep things clean. I also fixed some clippy lints as of 1.74.1. No functional changes intended on either commit.

### CODE REVIEW

**Review by w-henderson — COMMENTED** [review]
Thanks for this!

I think my (somewhat loosely-held) opinion is that we shouldn't change the edition of the existing crates unless we are updating them too with at least some bug fix or something, so my preference is to set `resolver="1"` in the workspace configuration, unless you feel strongly otherwise?

Thanks for fixing all the lints - I too have my editor configured to be as annoying as possible but I hadn't got around to it!

**Comment by Grinkers** [comment]
Personal opinion, but I feel for actively developed things `rust stable - 2 versions` (1.74 is stable today, so supporting rust 1.72) is good. 2 isn't a perfect number, but 1 feels too short, and 3 feels excessive.  Given the rest of the findings, I think this would be a good policy for now until we see what rust edition 2024 ends up with (potentially a lot of nice trait stuff, for more generic async things).

I had typed up a suggestion about maybe this stable crate being `rust edition - 1`, but this crate already doesn't work on editions that only have 'resolver = "1"`

a quick try on rust 1.36 (first 2018 edition) gives
```
Features and dependencies cannot have the same name: `tokio`
```

and 1.69 gives
```
namespaced features with the `dep:` prefix are only allowed on the nightly channel and requires the `-Z namespaced-features` flag on the command-line
```

https://github.com/foresterre/cargo-msrv
useful tool for looking into this.

I will let you decide on the final decision, but a little note somewhere in the docs/readme would be nice. If we go the route of having a MSRV, a CI test for it would also be nice.

**Comment by Grinkers** [comment]
force-push to disable a warning on tests, previously missed but spotted while doing the last post.

**Comment by w-henderson** [comment]
Running `cargo-msrv` tells me that the MSRV at the moment is 1.60, have I done it right and is this something we actually want to arbitrarily deviate from and explicitly not support versions that will work just fine?

**Comment by Grinkers** [comment]
1.60 is also what I got (on at least x86_64 linux). At least the way I see it, Minimal Supported Rust Version sort of implies we would consider it a bug if it didn't work on the MSRV, but it may work on older versions. Saying `- 2 versions` is just saying that's all that is guaranteed (in this case, 1.60 would be reality).

As far as resolver and edition 2021 goes, that's 1.56 and up. So unless work was done to support older versions of rust, I don't think there's any harm in bumping these.

I'll leave the policy up to you, but it would be nice to know for everybody. I just suspect it'll be hard to keep really old versions as the MSRV with tokio and async/traits. Presumably there's a lot of async/trait stuff coming in 2024. One would assume tokio would jump on it relatively quickly, as I think they're one of the main driving forces behind it.

**Comment by Grinkers** [comment]
https://github.com/tokio-rs/tokio#supported-rust-versions
https://github.com/tokio-rs/tokio/blob/master/.github/workflows/ci.yml#L28

A couple useful links relating to the discussion. 

I suspect without the tokio feature, this crate could support 1.36.

**Comment by w-henderson** [comment]
I think for now we say MSRV 1.60 and then bump it if we need/want to support new stuff in the future. Can you make a CI test for that too?

**Comment by Grinkers** [comment]
I added a CI test. You can see it running on my fork

https://github.com/Grinkers/Humphrey/actions

the "test" version had 1.59.0 set (which should fail) and then bumping it up to 1.60.0 passes.
https://github.com/Grinkers/Humphrey/commits/master/ commits here. this PR has it nicely squashed for clean history.

It seems like at least one example is broken though `cargo check --manifest-path examples/auth/Cargo.toml` also fails on my local machine with upstream/master checked out (cee1a6f9b419e3ab3c288457e0896f1b874c8802). So I think it's unrelated to the changes in this PR. As mentioned in #108, `cargo build --features tokio` also is failing on master.

**Comment by w-henderson** [comment]
Humphrey Server doesn't support the `tokio` feature so that's probably why the whole thing doesn't build with it enabled - that's a task for some point in the future. I'll sort out the auth example after we've merged this - could you just add something to the README indicating the MRSV? Then I think we're good to merge.

**Comment by Grinkers** [comment]
Happy new year!

Added a note. I kept it vague while the crate is still under development (under promise and over deliver). We'll see what rust 2024 ends up being!

**Review by w-henderson — APPROVED** [review]
This looks good, thank you!

**Pull request merged**
